python/cudf/cudf/core/index.py

```diff
@@ -1411,24 +1466,196 @@ def __cuda_array_interface__(self):
         return self._values.__cuda_array_interface__
 
 
+class BaseNumericIndex(GenericIndex):
+    """Immutable, ordered and sliceable sequence of labels.
+    The basic object storing row labels for all cuDF objects.
+
+    Parameters:
+    -----------
+    data : array-like (1-dimensional)
+    dtype : NumPy dtype,
+            but not used.
+    copy : bool
+        Make a copy of input data.
+    name : object
+        Name to be stored in the index.
+
+    Returns
+    -------
+    Index
+    """
+
+    def __new__(cls, data=None, dtype=None, copy=False, name=None):
+
+        out = Frame().__new__(cls)
+        dtype = index_to_dtype[cls]
+        if copy:
+            data = column.as_column(data, dtype=dtype).copy()
+
+        kwargs = _setdefault_name(data, name=name)
+
+        data = column.as_column(data, dtype=dtype)
+
+        out._initialize(data, **kwargs)
+
+        return out
+
+
+class Int8Index(BaseNumericIndex):
+    __doc__ = BaseNumericIndex.__doc__
+
+    def __init__(cls, data=None, dtype=None, copy=False, name=None):
+        pass
+
+
+class Int16Index(BaseNumericIndex):
+    __doc__ = BaseNumericIndex.__doc__
```

Comment:
What do you think about using https://github.com/rapidsai/cudf/blob/branch-0.15/python/cudf/cudf/utils/docutils.py#L58 here? If we decide to change the copying behaviour for docstrings, we'll only have to do it in that one place